### PR TITLE
feat: I2C and Notecard mutex callbacks

### PR DESCRIPTION
**Pull request by zfields**

### CODE REVIEW

File: src/Notecard.cpp [hunk 1]
```diff
@@ -404,72 +383,126 @@ J *Notecard::requestAndResponseWithRetry(J *req, uint32_t timeoutSeconds)
...
 }
 
 /**************************************************************************/
 /*!
-    @brief  Determines if there is an error string present in a response object.
-    @param    rsp
-              A `J` JSON Response object.
-    @return `True` if the response object contains an error.
+    @brief  Set the lock/unlock functions the Notecard uses for I2C access.
+    @param    lockI2cFn
```
Comment (haydenroche5) [inline src/Notecard.cpp:455]: Should this read lockI2cFn_ (with the trailing underscore)? Also, is the trailing underscore your way of indicating that the variable is a function parameter?

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: src/Notecard.cpp (hunk 1 above)
Comment (zfields) [reply src/Notecard.cpp:455]: > Should this read lockI2cFn_ (with the trailing underscore)?

That would be to taste. I don't put the trailing underscore in the `.h` file, which is typically where I put Doxygen comments.

> Also, is the trailing underscore your way of indicating that the variable is a function parameter?

Yes.

File: src/Notecard.cpp [hunk 2]
```diff
@@ -404,72 +383,126 @@ J *Notecard::requestAndResponseWithRetry(J *req, uint32_t timeoutSeconds)
...
-bool Notecard::responseError(J *rsp)
-{
-    return NoteResponseError(rsp);
+void Notecard::setFnI2cMutex(mutexFn lockI2cFn_, mutexFn unlockI2cFn_) {
+    NoteSetFnI2CMutex(lockI2cFn_, unlockI2cFn_);
+}
+
+/**************************************************************************/
+/*!
+    @brief  Set the lock/unlock functions the host MCU uses to ensure
```
[the remark this answers is not in the document]
Comment (zfields) [reply src/Notecard.cpp:470]: Ughhh... I agree.

However, I'm not sure an example should hold up the release. I can make a high-priority task to create `Example8_MultithreadedCommunication.ino` (or some such thing).

Does this seem like a reasonable compromise?

[the remark this answers is not in the document]
Comment (haydenroche5) [reply src/Notecard.cpp:470]: Discussed offline.

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**